lib/get-specs-from-issue.js

```diff
@@ -1,39 +1,93 @@
+const _ = require('lodash');
+
 module.exports = getSpecsFromIssue;
 
 function getSpecsFromIssue(context, issueNumber) {
   console.log('started getting specs out of issue');
-  let issue = context.github.issues;
-  let specs = [];
+  const repo = context.repo();
 
   return Promise.resolve()
-  .then(() => Promise.all([
+    .then(() => getIssueCompleteText(issueNumber, repo))
+    .then(issueCompleteText => getFirstSpecsBlock(issueCompleteText))
+    .then(specsBlock => getSpecsListItems(specsBlock));
+}
+
+function getIssueCompleteText(issueNumber, repo) {
+  return Promise.all([
     context.github.issues.get(...repo, issueNumber),
     context.github.issues.listComments(...repo, issueNumber)
-  ]))
-  .then(data => {
-    const { body, comments } = data;
-    const allCommentsBody = comments
-      .map((comment) => comment.body)
-      .push(body);
+  ])
+    .then((issueDataResponse) => {
+      const [issueBody, issueComments] = issueDataResponse;
+
+      return issueComments
+        .map((comment) => comment.body)
+        .unshift(issueBody)
+        .join('\n');
+    });
+}
+
+function getFirstSpecsBlock(issueCompleteText) {
+  const specsBlockStartRegex = /^\s*?(#+)\s*?specs/gi;
+  var isInsideSpecsBlock = false;
+  var specsBlockTopLevelHeader;
+  const specsBlock = [];
 
-    pushSpecs(specs, allCommentsBody);
+  _.forEach(issueCompleteText.split('\n'), line => {
+    const lineMatchResult = specsBlockStartRegex.exec(line);
 
-    return specs;
+    if (!isInsideSpecsBlock && lineMatchResult) {
+      isInsideSpecsBlock = true;
+      specsBlockTopLevelHeader = `${lineMatchResult[1]}#`;
+      return;
+    }
+
+    const isValidLine = !isMarkdownTitle(line) ||
+      isValidMarkdownTitle(line, specsBlockTopLevelHeader);
+
+    if (!isValidLine && isInsideSpecsBlock) {
+      return false;
+    }
+
+    if (isValidLine && isInsideSpecsBlock) {
+      specsBlock.push(line);
+      return;
+    }
   });
+
+  return specsBlock;
+}
```

Comment:
Some Questions:
 - Why do we need to `trim()` here
   > Do only place we need it that I can think of is where we check `startsWith`. The rest are regex matches with capturing groups that will lose the `whitespaces` for us.
- Your `isLineValid` will not pass the case of some text
   > In my opinion, the only case of invalid line is a higher title.

Reply:
>The only place we need it that I can think of is where we check startsWith. The rest are regex matches with capturing groups that will lose the whitespaces for us.
but you are using `startWith` here. for checking if the line is a title and is smaller than our title.
It can also save us all of the white space checks inside the regex which is far more expensive than trimming and checking `startsWith`. It also means you can lose all the whitespace checks in all the regex which is checked for each one of them separately.

> In my opinion, the only case of invalid line is a higher title.
That's not true as we defined this mission. later, we might add more stuff, but currently, that will introduce more problems than benefits.